app/src/main/java/io/zenandroid/onlinego/ui/screens/puzzle/PuzzleDirectoryUI.kt

```diff
@@ -0,0 +1,432 @@
+                                                                Column(modifier = Modifier.padding(8.dp)) {
+                                                                    Text(
+                                                                        text = it.name,
+                                                                        style = TextStyle.Default.copy(
+                                                                            fontSize = 12.sp,
+                                                                            fontWeight = FontWeight.Bold
+                                                                        )
+                                                                    )
+                                                                    it.owner?.let {
+                                                                        val flag = convertCountryCodeToEmojiFlag(it.country)
+                                                                        val ago = getRelativeTimeSpanString((ts ?: now()).toEpochMilli())
+                                                                        Text(
+                                                                            text = "by ${it.username} $flag - visited $ago",
+                                                                            maxLines = 1,
+                                                                            style = TextStyle.Default.copy(
+                                                                                fontSize = 9.sp,
+                                                                                fontWeight = FontWeight.Light
+                                                                            )
+                                                                        )
+                                                                    }
+                                                                }
+                                                            }
+                                                        }
+                                                    }
+                                                } ?: Surface(
+                                                    shape = MaterialTheme.shapes.medium,
+                                                    modifier = Modifier.height(50.dp)
+                                                ) {}
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            state.collections.map { it.value }.ifEmpty { null }?.let { collections ->
+                item {
+                    var compositionCount by remember { mutableStateOf(0) }
+                    SideEffect { compositionCount++ }
```

Comment:
@bqv FYI: this doesn't work as increasing the compositionCount causes a recomposition, which in turn causes a new recomposition (as you're reading compositionCount in the same scope) which causes the SideEffect to trigger, which causes the recomposition to occur, etc... This is a rather interesting case of backwards write: https://developer.android.com/jetpack/compose/side-effects#sideeffect-publish

Reply:
ah! I wondered what was happening. I actually left that in by accident, it was me trying to debug something